### DISCUSSION TITLE
./world/session.lock issue on DSM 7.2

### DISCUSSION THREAD

**Discussion by LilVapers** [itzg/docker-minecraft-server#2710] [Q&A]
This is the error log. Any suggestions on how to fix it? As far as I can tell its something to do with session.lock.

2024/03/11 19:05:03 | stdout | 2024-03-11T23:05:03.194Z	INFO	mc-server-runner	Done
-- | -- | --
2024/03/11 19:05:03 | stdout | at net.fabricmc.installer.ServerLauncher.main(ServerLauncher.java:69) ~[fabric-server-mc.1.20.1-loader.0.15.7-launcher.1.0.0.jar:1.0.0]
2024/03/11 19:05:03 | stdout | at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:69) ~[fabric-loader-0.15.7.jar:?]
2024/03/11 19:05:03 | stdout | at net.fabricmc.loader.impl.launch.knot.KnotServer.main(KnotServer.java:23) ~[fabric-loader-0.15.7.jar:?]
2024/03/11 19:05:03 | stdout | at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) ~[fabric-loader-0.15.7.jar:?]
2024/03/11 19:05:03 | stdout | at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470) ~[fabric-loader-0.15.7.jar:?]
2024/03/11 19:05:03 | stdout | at net.minecraft.server.Main.main(Main.java:136) ~[server-intermediary.jar:?]
2024/03/11 19:05:03 | stdout | at net.minecraft.class_32.method_52236(class_32.java:377) ~[server-intermediary.jar:?]
2024/03/11 19:05:03 | stdout | at net.minecraft.class_32$class_5143.<init>(class_32.java:398) ~[server-intermediary.jar:?]
2024/03/11 19:05:03 | stdout | at net.minecraft.class_5125.method_26803(class_5125.java:34) ~[server-intermediary.jar:?]
2024/03/11 19:05:03 | stdout | at java.nio.channels.FileChannel.open(Unknown Source) ~[?:?]
2024/03/11 19:05:03 | stdout | at java.nio.channels.FileChannel.open(Unknown Source) ~[?:?]
2024/03/11 19:05:03 | stdout | at sun.nio.fs.UnixFileSystemProvider.newFileChannel(Unknown Source) ~[?:?]
2024/03/11 19:05:03 | stdout | at sun.nio.fs.UnixException.rethrowAsIOException(Unknown Source) ~[?:?]
2024/03/11 19:05:03 | stdout | at sun.nio.fs.UnixException.rethrowAsIOException(Unknown Source) ~[?:?]
2024/03/11 19:05:03 | stdout | at sun.nio.fs.UnixException.translateToIOException(Unknown Source) ~[?:?]
2024/03/11 19:05:03 | stdout | java.nio.file.AccessDeniedException: ./world/session.lock
2024/03/11 19:05:03 | stdout | [23:05:03] [main/ERROR]: Failed to start the minecraft server
2024/03/11 19:05:02 | stdout | [23:05:02] [main/INFO]: Environment: authHost='https://authserver.mojang.com', accountsHost='https://api.mojang.com', sessionHost='https://sessionserver.mojang.com', servicesHost='https://api.minecraftservices.com', name='PROD'
2024/03/11 19:04:50 | stdout | [23:04:50] [main/INFO]: SpongePowered MIXIN Subsystem Version=0.8.5 Source=file:/data/libraries/net/fabricmc/sponge-mixin/0.12.5+mixin.0.8.5/sponge-mixin-0.12.5+mixin.0.8.5.jar Service=Knot/Fabric Env=SERVER
2024/03/11 19:04:50 | stdout | - minecraft 1.20.1
2024/03/11 19:04:50 | stdout | - java 17
2024/03/11 19:04:50 | stdout | \-- mixinextras 0.3.5
2024/03/11 19:04:50 | stdout | - fabricloader 0.15.7
2024/03/11 19:04:50 | stdout | [23:04:50] [main/INFO]: Loading 4 mods:
2024/03/11 19:04:49 | stdout | [23:04:49] [main/INFO]: Loading Minecraft 1.20.1 with Fabric Loader 0.15.7
2024/03/11 19:04:48 | stdout | Starting net.fabricmc.loader.impl.game.minecraft.BundlerClassPathCapture
2024/03/11 19:04:46 | stdout | [init] Starting the Minecraft server...
2024/03/11 19:04:44 | stdout | [init] Setting initial memory to 1G and max to 14G
2024/03/11 19:04:37 | stdout | [mc-image-helper] 23:04:37.858 INFO  : Created/updated 1 property in /data/server.properties
2024/03/11 19:04:33 | stdout | [mc-image-helper] 23:04:33.584 INFO  : Fabric launcher for minecraft 1.20.1 loader 0.15.7 is already available
2024/03/11 19:04:28 | stdout | [init] Resolving type given FABRIC
2024/03/11 19:04:28 | stdout | [init] Running as uid=1000 gid=1000 with /data as 'drwxr-xr-x 1 1000 1000 522 Mar  9 23:16 /data'

**Comment by itzg** [comment]
Correct, the `session.lock` is the problem:

```
java.nio.file.AccessDeniedException: ./world/session.lock
```

Since it is "access denied" check that the file ownership of the world files are all owned by uid=1000 and that the world directory itself is also owned by uid=1000 and writable.

**Reply by itzg** [reply]
Check the ownership and permissions use the `-d` option of `ls`, such as:

```
root@e922b64856bb:/data# ls -ldn world
drwxrwxr-x 5 1000 1000 4096 Mar 12 00:05 world
```

(I also used `-n` to see numeric ownership IDs)

**Comment by LilVapers** [comment]
Hey, thanks for getting back to me. I added a screenshot of the current variables I am using (UID 1000 and GID 1000). I am a bit of a noob at this, I am not sure how to do "ls -ldn world" on the terminal when the server is starting. I can't type anything there etc.

![Screenshot 2024-03-11 200620](https://github.com/itzg/docker-minecraft-server/assets/88253560/cc429e9d-f50c-43ef-bc0b-84a051d7f126)

**Reply by itzg** [reply]
I meant run a Linux `ls` command in the DSM's shell. Looking in their file browser is probably sufficient to inspect ownership and file modes. I don't know DSM systems so I can't offer much help there.
